### PR TITLE
Cleanup ats_tcp_somaxconn to use std::ifstream

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
Use `std::ifstream` and avoid dependency on `TextBuffer` with cleaner code.

### CODE REVIEW

File: lib/ts/ink_inet.cc [hunk 1]
```diff
@@ -661,19 +661,19 @@ ats_tcp_somaxconn()
 #endif
 
   fd = open("/proc/sys/net/ipv4/tcp_max_syn_backlog", O_RDONLY);
-  if (fd != -1) {
-    TextBuffer text(0);
-    text.slurp(fd);
-    if (!text.empty()) {
-      value = strtoul(text.bufPtr(), nullptr, 10);
+  if (fd != ts::NO_FD) {
```
Comment (bryancall) [inline lib/ts/ink_inet.cc]: It would be simpler to use ifstream.

Comment (SolidWallOfCode) [reply lib/ts/ink_inet.cc]: Like
```
std::ifstream f("/proc/sys/net/ipv4/tcp_max_syn_backlog", std::ifstream::in);
if (f.good()) {
  value << f;
}
```

Comment (bryancall) [reply lib/ts/ink_inet.cc]: Yes

**Comment by SolidWallOfCode** [comment]
Don't approve yet, I need to fix the commit message too but need to wait for CI to finish.

**Pull request merged**